src/AutoRest.CSharp/Mgmt/Decorator/ExtensionDetection.cs

```diff
@@ -1,24 +1,39 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License
 
+using System.Collections.Concurrent;
+using AutoRest.CSharp.AutoRest.Plugins;
 using AutoRest.CSharp.Input;
 
 namespace AutoRest.CSharp.Mgmt.Decorator
 {
     internal static class ExtensionDetection
     {
-        public static bool IsExtension(OperationGroup operationGroup)
+        private static ConcurrentDictionary<string, bool> _valueCache = new ConcurrentDictionary<string, bool>();
```

Comment:
I would recommend not making this static. We don't currently do it but it's possible that multiple swaggers would be processed in the same process. This might also make the in-proc generator running tests tricky.

Reply:
Recommend putting this on management BuildContext and passing that around.